libraries/AP_Math/AP_Math.cpp

```diff
@@ -24,16 +24,17 @@ template bool is_equal<double>(const double v_1, const double v_2);
 template <class T>
 float safe_asin(const T v)
 {
-    if (isnan(static_cast<float>(v))) {
+    float f = v;
```

Comment:
Again, *safe_asin* is instantiated for doubles (I'm not sure if it is actually used), so I think this doesn't really improve the situation.